compiler/noirc_evaluator/src/ssa/opt/remove_if_else.rs

```diff
@@ -1,3 +1,94 @@
+//! This file contains the SSA `remove_if_else` pass - a required pass for ACIR to remove any remaining
+//! `Instruction::IfElse` in the singular program-function,  and remplace them with
+//! arithmetic operations using the `then_condition`.
+//!
+//! ACIR/Brillig differences within this pass:
+//!   - This pass is strictly ACIR-only and never mutates brillig functions.
+//!
+//!
+//! Conditions:
+//!   - Precondition: Flatten CFG has been performed which should result in a program having only
+//!     one basic block, representing the `main` function.
```

Comment:
Each `#[fold]` function gets flattened as well sso it's more accurate to say we get 1 block for each ACIR function in the program. Where we expect to have `main` (if constrained) along with any constrained function marked `#[fold]`.

Reply:
Done